–ø–æ—Å–ª–µ —Ç–æ–≥–æ –∫–∞–∫ –º—ã —Å–∫–∞—á–∞–ª–∏ –¥–∞–Ω—ã–Ω–µ –Ω–∞–º –Ω—É–∂–Ω–æ —Å–¥–µ–ª–∞—Ç—å –≤—Å–µ –≤–æ—Ç —ç—Ç–æ: 

The raw reads were aligned using bwa mem to the human genome reference (hg19 or hg38) corresponding to the reference used for the true set of somatic variants for each dataset [26]. Duplicated reads were marked with sambamba, and base quality score recalibration (BQSR) was done with GATK4 [27, 28]. Influence of post-alignment procedures on variant calling was evaluated using four different strategies (bwa; bwa‚Äâ+‚Äâdeduplication; bwa‚Äâ+‚ÄâBQSR; bwa‚Äâ+‚Äâdeduplication‚Äâ+‚ÄâBQSR).
---

## 1. –í—ã—Ä–∞–≤–Ω–∏–≤–∞–Ω–∏–µ reads —Å –ø–æ–º–æ—â—å—é `bwa mem`

–ü–µ—Ä–µ–¥ –Ω–∞—á–∞–ª–æ–º –≤—ã—Ä–∞–≤–Ω–∏–≤–∞–Ω–∏—è –Ω–µ–æ–±—Ö–æ–¥–∏–º–æ –ø—Ä–æ–∏–Ω–¥–µ–∫—Å–∏—Ä–æ–≤–∞—Ç—å —Ä–µ—Ñ–µ—Ä–µ–Ω—Å–Ω—ã–π –≥–µ–Ω–æ–º.

### üîπ –®–∞–≥ 1: –ò–Ω–¥–µ–∫—Å–∞—Ü–∏—è —Ä–µ—Ñ–µ—Ä–µ–Ω—Å–Ω–æ–≥–æ –≥–µ–Ω–æ–º–∞

```bash
bwa index /media/ivan/KINGSTON/self_project/reference/GCF_000001405.40_GRCh38.p14_genomic.fna
```

#### –í—ã–≤–æ–¥ –ø–æ—Å–ª–µ –∏–Ω–¥–µ–∫—Å–∏—Ä–æ–≤–∞–Ω–∏—è:

```
[bwt_gen] Finished constructing BWT in 728 iterations.
[bwa_index] 2133.71 seconds elapse.
[bwa_index] Update BWT... 16.51 sec
[bwa_index] Pack forward-only FASTA... 17.75 sec
[bwa_index] Construct SA from BWT and Occ... 1369.41 sec
[main] Version: 0.7.17-r1188
[main] CMD: bwa index /media/ivan/KINGSTON/self_project/reference/GCF_000001405.40_GRCh38.p14_genomic.fna
[main] Real time: 3993.403 sec; CPU: 3560.819 sec
```

---

### üîπ –®–∞–≥ 2: –í—ã—Ä–∞–≤–Ω–∏–≤–∞–Ω–∏–µ —á—Ç–µ–Ω–∏–π —Å –ø–æ–º–æ—â—å—é `bwa mem`

```bash
bwa mem -t 8 /media/ivan/KINGSTON/self_project/reference/GCF_000001405.40_GRCh38.p14_genomic.fna read1.fastq.gz read2.fastq.gz | samtools view -Sb - > aligned.bam
```
—Ä–µ–∞–ª—å–Ω–∞—è –∫–æ–º–∞–Ω–¥–∞ 
(base) ivan@ivan-Redmi-Book-Pro-15-2022:/media/ivan/KINGSTON/self_project/bams$ bwa mem -t 12 /media/ivan/KINGSTON/self_project/reference/GCF_000001405.40_GRCh38.p14_genomic.fna /media/ivan/KINGSTON/self_project/raw_data/SEQC2/SRR7890879.sralite.1_1.fastq.gz /media/ivan/KINGSTON/self_project/raw_data/SEQC2/SRR7890879.sralite.1_2.fastq.gz | samtools view -b - > SRR7890879_aligned.bam && { echo -e "\a"; notify-send "BWA Index" "–£—Ä–∞ –Ω–∞—Ñ–∏–≥!"; }

–≤—ã–≤–æ–¥ –∫–æ–º–∞–Ω–¥—ã
(base) ivan@ivan-Redmi-Book-Pro-15-2022:/media/ivan/KINGSTON/self_project/bams$ bwa mem -t 12 /media/ivan/KINGSTON/self_project/reference/GCF_000001405.40_GRCh38.p14_genomic.fna /media/ivan/KINGSTON/self_project/raw_data/SEQC2/SRR7890879.sralite.1_1.fastq.gz /media/ivan/KINGSTON/self_project/raw_data/SEQC2/SRR7890879.sralite.1_2.fastq.gz | samtools view -b - > SRR7890879_aligned.bam && { echo -e "\a"; notify-send "BWA Index" "–£—Ä–∞ –Ω–∞—Ñ–∏–≥!"; }


[main] Version: 0.7.17-r1188
[main] CMD: bwa mem -t 12 /media/ivan/KINGSTON/self_project/reference/GCF_000001405.40_GRCh38.p14_genomic.fna /media/ivan/KINGSTON/self_project/raw_data/SEQC2/SRR7890879.sralite.1_1.fastq.gz /media/ivan/KINGSTON/self_project/raw_data/SEQC2/SRR7890879.sralite.1_2.fastq.gz
[main] Real time: 1672.178 sec; CPU: 17157.415 sec

* `-t 8` ‚Äî –∏—Å–ø–æ–ª—å–∑–æ–≤–∞—Ç—å 8 –ø–æ—Ç–æ–∫–æ–≤
* `samtools view -Sb -` ‚Äî –∫–æ–Ω–≤–µ—Ä—Ç–∞—Ü–∏—è SAM –≤ BAM –∏ –≤—ã–≤–æ–¥ –≤ —Ñ–∞–π–ª `aligned.bam`

2. –£–¥–∞–ª–µ–Ω–∏–µ –¥—É–±–ª–∏–∫–∞—Ç–æ–≤ —Å –ø–æ–º–æ—â—å—é sambamba markdup
bash
conda create -n sambamba_env -c bioconda sambamba --y
conda activate sambamba_env
--y –∞–≤—Ç–æ–º–∞—Ç–∏—á–µ—Å–∫–∏–π –æ—Ç–≤–µ—Ç –¥–∞ –Ω–∞ –∑–∞–ø—Ä–æ—Å—ã

sambamba markdup -t 12 SRR7890879_aligned.bam SRR7890879_aligned_deduplicated.bam
—Ä–µ–∞–ª—å–Ω–∞—è –∫–æ–º–∞–Ω–¥–∞ sambamba markdup -t 12 SRR7890879_aligned.bam SRR7890879_aligned_deduplicated.bam
–ì–¥–µ:

    -t 8 ‚Äî –∫–æ–ª–∏—á–µ—Å—Ç–≤–æ –ø–æ—Ç–æ–∫–æ–≤.

    deduplicated.bam ‚Äî –≤—ã—Ö–æ–¥–Ω–æ–π —Ñ–∞–π–ª –±–µ–∑ –¥—É–±–ª–∏–∫–∞—Ç–æ–≤.

–≤—ã–≤–æ–¥ –∫–æ–º–∞–Ω–¥—ã 
finding positions of the duplicate reads in the file...
  sorted 36352820 end pairs
     and 38948 single ends (among them 0 unmatched pairs)
  collecting indices of duplicate reads...   done in 3910 ms
  found 15783533 duplicates
collected list of positions in 2 min 36 sec
marking duplicates...
total time elapsed: 7 min 55 sec

4. –ë–∞–∑–æ–≤–æ–µ–≤–∞—è —Ä–µ–∫–∞–ª–∏–±—Ä–æ–≤–∫–∞ –∫–∞—á–µ—Å—Ç–≤–∞ (BQSR) —Å –ø–æ–º–æ—â—å—é GATK4
a. –ì–µ–Ω–µ—Ä–∞—Ü–∏—è —Ç–∞–±–ª–∏—Ü—ã —Ä–µ–∫–∞–ª–∏–±—Ä–æ–≤–∫–∏:
bash

gatk BaseRecalibrator \
   -I deduplicated.bam \
   -R /path/to/hg19.fasta \
   --known-sites /path/to/dbsnp.vcf.gz \
   --known-sites /path/to/1000G_phase1.snps.high_confidence.hg19.vcf \
   -O recal_data.table

b. –ü—Ä–∏–º–µ–Ω–µ–Ω–∏–µ —Ä–µ–∫–∞–ª–∏–±—Ä–æ–≤–∫–∏:
bash

gatk ApplyBQSR \
   -R /path/to/hg19.fasta \
   -I deduplicated.bam \
   --bqsr-recal-file recal_data.table \
   -O recalibrated.bam

–ì–¥–µ:

    --known-sites ‚Äî —Ñ–∞–π–ª—ã —Å –∏–∑–≤–µ—Å—Ç–Ω—ã–º–∏ –≤–∞—Ä–∏–∞–Ω—Ç–∞–º–∏ (–Ω–∞–ø—Ä–∏–º–µ—Ä, dbSNP).

    recalibrated.bam ‚Äî –∏—Ç–æ–≥–æ–≤—ã–π BAM-—Ñ–∞–π–ª –ø–æ—Å–ª–µ —Ä–µ–∫–∞–ª–∏–±—Ä–æ–≤–∫–∏.





---


---

### –ò–Ω—Å—Ç—Ä—É–º–µ–Ω—Ç—ã –¥–ª—è –∞–Ω–∞–ª–∏–∑–∞ —Å–æ–º–∞—Ç–∏—á–µ—Å–∫–∏—Ö –º—É—Ç–∞—Ü–∏–π
1. **Mutect2 (GATK)**  
   - –ü–æ–ø—É–ª—è—Ä–Ω—ã–π –∏–Ω—Å—Ç—Ä—É–º–µ–Ω—Ç –æ—Ç Broad Institute.  
   - –ò—Å–ø–æ–ª—å–∑—É–µ—Ç Gradient Boosting (XGBoost-like) –¥–ª—è —Ñ–∏–ª—å—Ç—Ä–∞—Ü–∏–∏ –ª–æ–∂–Ω—ã—Ö –≤–∞—Ä–∏–∞–Ω—Ç–æ–≤.
     
```bash
#Mutect2
export OMP_NUM_THREADS={resources.cpus_per_task}
file_name=$(basename {input.paired[1]})
name=${{file_name%.sort*}}
gatk Mutect2 \
	--java-options -Xmx8G \
	--native-pair-hmm-threads 8 \
	--reference {input.ref} \
	-I {input.paired[0]} \
 	-I {input.paired[1]} \
 	-normal $name \
 	--germline-resource {input.gnomad} \
 	--genotype-germline-sites true \
 	--genotype-pon-sites true \
 	--interval-padding {params.interval_padding} \
 	-pon {input.pon} \
 	-L {input.target} \
 	--f1r2-tar-gz {output.f1r2} \
 	-O {output.vcf}
```

2. **Strelka**  
   - –í—ã—Å–æ–∫–∞—è —Ç–æ—á–Ω–æ—Å—Ç—å –¥–ª—è SNV –∏ –∏–Ω–¥–µ–ª–æ–≤.  
   - –ü—Ä–∏–º–µ–Ω—è–µ—Ç HMM –∏ ML-—Ñ–∏–ª—å—Ç—Ä—ã.
  
`     #Strelka
configureStrelkaSomaticWorkflow.py ‚Äìexome \
	--normal={input.paired[1]} \
	--tumor={input.paired[0]} \
	--ref={input.ref} \
	--runDir=tmp \
	--callRegions {input.target}.gz`

3. **DeepSomatic**  
   - CNN + LSTM –¥–ª—è –ø—Ä–µ–¥—Å–∫–∞–∑–∞–Ω–∏—è –º—É—Ç–∞—Ü–∏–π.  
`#DeepSomatic
run_deepsomatic \
	--model_type=WGS \
	--ref={input.ref} \
	--reads_normal={input.paired[1]} \
	--reads_tumor={input.paired[0]} \
	--output_vcf=DeepSomatic.somatic.vcf.gz \
	--sample_name_tumor="{wildcards.seq_name}" \
	--sample_name_normal="{wildcards.normal}" \
	--num_shards=20 \
	--logging_dir=DeepSomatic \
	--intermediate_results_dir=DeepSomatic/tmp \
        --regions={input.target}`
4. **VarNet (Google Health)**  
   - –û—Å–Ω–æ–≤–∞–Ω –Ω–∞ —Å–≤–µ—Ä—Ç–æ—á–Ω—ã—Ö —Å–µ—Ç—è—Ö (CNN).  
`#VarNet
python /VarNet/filter.py \
	--sample_name {wildcards.seq_name} \
	--normal_bam {input.paired[1]} \
	--tumor_bam {input.paired[0]} \
	--processes 8 \
	--output_dir VarNet \
	--reference {input.ref} \
	--region_bed {input.target}
python /VarNet/predict.py \
	--sample_name {wildcards.seq_name} \
	--normal_bam {input.paired[1]} \
	--tumor_bam {input.paired[0]} \
	--processes 8 \
	--output_dir VarNet \
	--reference {input.ref}
`
5. **VarDict**  
   - –≠—Ñ—Ñ–µ–∫—Ç–∏–≤–µ–Ω –¥–ª—è –≥–µ—Ç–µ—Ä–æ–≥–µ–Ω–Ω—ã—Ö –æ–ø—É—Ö–æ–ª–µ–π.  
   - –ò—Å–ø–æ–ª—å–∑—É–µ—Ç Random Forest.
 `#VarDict
 VarDict
	‚Äìdedup \
	-th 8 \
	-G {input.ref} \
	-f 0.02 \
	-N {wildcards.seq_name} \
	-r 3 \
	-b "{input.paired[0]}|{input.paired[1]}" \
	-c 1 -S 2 -E 3 -g 4 {input.target} | testsomatic.R | var2vcf_paired.pl -N "{wildcards.seq_name}|{wildcards.normal}" -f 0.02 > {output}`





___
—Ä—É–±—Ä–∏–∫–∞ –≤–æ–ø—Ä–æ—Å—ã 
![image](https://github.com/user-attachments/assets/d7ab7f04-4c0f-4b0c-b591-3c59519bcfd7)
–ø–æ—á–µ–º—É —ç—Ç–∏ —Ä–µ–±—è—Ç–∞ –Ω–µ —É–∫–∞–∑—ã–≤–∞—é—Ç –≤–µ—Ä—Å–∏–∏, –Ω–∞–ø—Ä–∏–º–µ—Ä sambamba –∏–ª–∏ bwa mem


     
