### Ğ—Ğ°Ğ³Ñ€ÑƒĞ·ĞºĞ° Ğ¸ Ğ¾Ğ±Ñ€Ğ°Ğ±Ğ¾Ñ‚ĞºĞ° Ğ´Ğ°Ğ½Ğ½Ñ‹Ñ…

#### 1. Ğ¡ĞºĞ°Ñ‡Ğ¸Ğ²Ğ°Ğ½Ğ¸Ğµ ÑÑ‹Ñ€Ñ‹Ñ… Ğ¿Ñ€Ğ¾Ñ‡Ñ‚ĞµĞ½Ğ¸Ğ¹ (FASTQ)
Ğ¡Ñ‹Ñ€Ñ‹Ğµ Ğ´Ğ°Ğ½Ğ½Ñ‹Ğµ (raw reads) Ğ´Ğ¾ÑÑ‚ÑƒĞ¿Ğ½Ñ‹ Ğ² NCBI SRA:  
ğŸ”— [SRR7890879](https://trace.ncbi.nlm.nih.gov/Traces/?view=run_browser&acc=SRR7890879&display=data-access)  

**ĞŸÑ€Ğ¸Ğ¼ĞµÑ‡Ğ°Ğ½Ğ¸Ğµ**: Ğ Ğ°Ğ·Ğ½Ğ¸Ñ†Ğ° Ğ¼ĞµĞ¶Ğ´Ñƒ Ñ„Ğ¾Ñ€Ğ¼Ğ°Ñ‚Ğ°Ğ¼Ğ¸ SRA: 
![image](https://github.com/user-attachments/assets/45a94084-ae0d-48ea-bfa2-cdfbd8dfe9f4)

- **SRA Lite** â€“ Ğ¾Ğ±Ğ»ĞµĞ³Ñ‡ĞµĞ½Ğ½Ğ°Ñ Ğ²ĞµÑ€ÑĞ¸Ñ Ğ±ĞµĞ· Ğ½ĞµĞºĞ¾Ñ‚Ğ¾Ñ€Ñ‹Ñ… Ğ¼ĞµÑ‚Ğ°Ğ´Ğ°Ğ½Ğ½Ñ‹Ñ….  
- **SRA Normalized** â€“ Ğ´Ğ°Ğ½Ğ½Ñ‹Ğµ Ğ½Ğ¾Ñ€Ğ¼Ğ°Ğ»Ğ¸Ğ·Ğ¾Ğ²Ğ°Ğ½Ñ‹ (Ğ½Ğ°Ğ¿Ñ€Ğ¸Ğ¼ĞµÑ€, Ğ¿Ğ¾ Ğ³Ğ»ÑƒĞ±Ğ¸Ğ½Ğµ Ğ¿Ğ¾ĞºÑ€Ñ‹Ñ‚Ğ¸Ñ).  

Ğ”Ğ»Ñ Ğ·Ğ°Ğ³Ñ€ÑƒĞ·ĞºĞ¸ FASTQ Ğ½Ğ°Ğ¿Ñ€ÑĞ¼ÑƒÑ (Ğ±ĞµĞ· Ğ¿Ñ€Ğ¾Ğ¼ĞµĞ¶ÑƒÑ‚Ğ¾Ñ‡Ğ½Ğ¾Ğ³Ğ¾ .sra) Ğ¸ÑĞ¿Ğ¾Ğ»ÑŒĞ·Ğ¾Ğ²Ğ°Ğ»Ğ°ÑÑŒ ĞºĞ¾Ğ¼Ğ°Ğ½Ğ´Ğ°:  
```bash
fasterq-dump --split-files --progress --threads 4 SRR7890883
```

#### 2. Ğ—Ğ°Ğ³Ñ€ÑƒĞ·ĞºĞ° Ğ´Ğ°Ñ‚Ğ°ÑĞµÑ‚Ğ° cancer-dream-syn3
```bash
mkdir -p cancer-dream-syn3/config cancer-dream-syn3/input cancer-dream-syn3/work
cd cancer-dream-syn3/config
wget https://raw.githubusercontent.com/bcbio/bcbio-nextgen/master/config/examples/cancer-dream-syn3.yaml
cd ../input
wget https://raw.githubusercontent.com/bcbio/bcbio-nextgen/master/config/examples/cancer-dream-syn3-getdata.sh
bash cancer-dream-syn3-getdata.sh
```

#### 3. Ğ ĞµÑ„ĞµÑ€ĞµĞ½ÑĞ½Ñ‹Ğ¹ Ğ³ĞµĞ½Ğ¾Ğ¼
Ğ¡ĞºĞ°Ñ‡Ğ°Ğ½ GRCh38 (GCF_000001405.40) Ğ¸Ğ· NCBI Datasets, Ñ‚Ğ°Ğº ĞºĞ°Ğº Ğ¾Ğ½ ÑƒĞºĞ°Ğ·Ğ°Ğ½ Ğ² Ñ€ĞµÑ„ĞµÑ€ĞµĞ½ÑĞ½Ñ‹Ñ… VCF:  
```bash
conda activate ncbi_datasets
datasets download genome accession GCF_000001405.40 --include genome
```
Ğ”Ğ»Ñ ÑĞºĞ¾Ğ½Ğ¾Ğ¼Ğ¸Ğ¸ Ğ¼ĞµÑÑ‚Ğ° Ğ¼Ğ¾Ğ¶Ğ½Ğ¾ Ğ¸ÑĞ¿Ğ¾Ğ»ÑŒĞ·Ğ¾Ğ²Ğ°Ñ‚ÑŒ `--dehydrated` Ñ Ğ¿Ğ¾ÑĞ»ĞµĞ´ÑƒÑÑ‰ĞµĞ¹ Ñ€ĞµĞ³Ğ¸Ğ´Ñ€Ğ°Ñ‚Ğ°Ñ†Ğ¸ĞµĞ¹:  
```bash
datasets rehydrate --directory ncbi_dataset
```

#### 4. ĞšĞ¾Ğ½Ğ²ĞµÑ€Ñ‚Ğ°Ñ†Ğ¸Ñ SRA Lite Ğ² FASTQ
```bash
conda activate sra_env
fastq-dump --split-files --gzip /path/to/SRR7890879.sralite.1
```
**Ğ ĞµĞ·ÑƒĞ»ÑŒÑ‚Ğ°Ñ‚**: 36,393,620 Ğ¿Ñ€Ğ¾Ñ‡Ñ‚ĞµĞ½Ğ¸Ğ¹.

#### 5. Ğ’Ñ‹Ñ€Ğ°Ğ²Ğ½Ğ¸Ğ²Ğ°Ğ½Ğ¸Ğµ Ğ½Ğ° Ñ€ĞµÑ„ĞµÑ€ĞµĞ½Ñ
Ğ˜ÑĞ¿Ğ¾Ğ»ÑŒĞ·Ğ¾Ğ²Ğ°Ğ½ `bwa mem` Ğ´Ğ»Ñ Ğ²Ñ‹Ñ€Ğ°Ğ²Ğ½Ğ¸Ğ²Ğ°Ğ½Ğ¸Ñ Ğ¿Ğ°Ñ€Ğ½Ñ‹Ñ… Ğ¿Ñ€Ğ¾Ñ‡Ñ‚ĞµĞ½Ğ¸Ğ¹:  
```bash
bwa mem -t 8 /path/to/hg38.fasta sample_1.fastq.gz sample_2.fastq.gz | samtools view -Sb - > aligned.bam
```

---

### Ğ˜Ğ½ÑÑ‚Ñ€ÑƒĞ¼ĞµĞ½Ñ‚Ñ‹ Ğ´Ğ»Ñ Ğ°Ğ½Ğ°Ğ»Ğ¸Ğ·Ğ° ÑĞ¾Ğ¼Ğ°Ñ‚Ğ¸Ñ‡ĞµÑĞºĞ¸Ñ… Ğ¼ÑƒÑ‚Ğ°Ñ†Ğ¸Ğ¹
1. **Mutect2 (GATK)**  
   - ĞŸĞ¾Ğ¿ÑƒĞ»ÑÑ€Ğ½Ñ‹Ğ¹ Ğ¸Ğ½ÑÑ‚Ñ€ÑƒĞ¼ĞµĞ½Ñ‚ Ğ¾Ñ‚ Broad Institute.  
   - Ğ˜ÑĞ¿Ğ¾Ğ»ÑŒĞ·ÑƒĞµÑ‚ Gradient Boosting (XGBoost-like) Ğ´Ğ»Ñ Ñ„Ğ¸Ğ»ÑŒÑ‚Ñ€Ğ°Ñ†Ğ¸Ğ¸ Ğ»Ğ¾Ğ¶Ğ½Ñ‹Ñ… Ğ²Ğ°Ñ€Ğ¸Ğ°Ğ½Ñ‚Ğ¾Ğ².
     
`#Mutect2
export OMP_NUM_THREADS={resources.cpus_per_task}
file_name=$(basename {input.paired[1]})
name=${{file_name%.sort*}}
gatk Mutect2 \
	--java-options -Xmx8G \
	--native-pair-hmm-threads 8 \
	--reference {input.ref} \
	-I {input.paired[0]} \
 	-I {input.paired[1]} \
 	-normal $name \
 	--germline-resource {input.gnomad} \
 	--genotype-germline-sites true \
 	--genotype-pon-sites true \
 	--interval-padding {params.interval_padding} \
 	-pon {input.pon} \
 	-L {input.target} \
 	--f1r2-tar-gz {output.f1r2} \
 	-O {output.vcf}`

2. **Strelka**  
   - Ğ’Ñ‹ÑĞ¾ĞºĞ°Ñ Ñ‚Ğ¾Ñ‡Ğ½Ğ¾ÑÑ‚ÑŒ Ğ´Ğ»Ñ SNV Ğ¸ Ğ¸Ğ½Ğ´ĞµĞ»Ğ¾Ğ².  
   - ĞŸÑ€Ğ¸Ğ¼ĞµĞ½ÑĞµÑ‚ HMM Ğ¸ ML-Ñ„Ğ¸Ğ»ÑŒÑ‚Ñ€Ñ‹.
  
`     #Strelka
configureStrelkaSomaticWorkflow.py â€“exome \
	--normal={input.paired[1]} \
	--tumor={input.paired[0]} \
	--ref={input.ref} \
	--runDir=tmp \
	--callRegions {input.target}.gz`

3. **DeepSomatic**  
   - CNN + LSTM Ğ´Ğ»Ñ Ğ¿Ñ€ĞµĞ´ÑĞºĞ°Ğ·Ğ°Ğ½Ğ¸Ñ Ğ¼ÑƒÑ‚Ğ°Ñ†Ğ¸Ğ¹.  
`#DeepSomatic
run_deepsomatic \
	--model_type=WGS \
	--ref={input.ref} \
	--reads_normal={input.paired[1]} \
	--reads_tumor={input.paired[0]} \
	--output_vcf=DeepSomatic.somatic.vcf.gz \
	--sample_name_tumor="{wildcards.seq_name}" \
	--sample_name_normal="{wildcards.normal}" \
	--num_shards=20 \
	--logging_dir=DeepSomatic \
	--intermediate_results_dir=DeepSomatic/tmp \
        --regions={input.target}`
4. **VarNet (Google Health)**  
   - ĞÑĞ½Ğ¾Ğ²Ğ°Ğ½ Ğ½Ğ° ÑĞ²ĞµÑ€Ñ‚Ğ¾Ñ‡Ğ½Ñ‹Ñ… ÑĞµÑ‚ÑÑ… (CNN).  
`#VarNet
python /VarNet/filter.py \
	--sample_name {wildcards.seq_name} \
	--normal_bam {input.paired[1]} \
	--tumor_bam {input.paired[0]} \
	--processes 8 \
	--output_dir VarNet \
	--reference {input.ref} \
	--region_bed {input.target}
python /VarNet/predict.py \
	--sample_name {wildcards.seq_name} \
	--normal_bam {input.paired[1]} \
	--tumor_bam {input.paired[0]} \
	--processes 8 \
	--output_dir VarNet \
	--reference {input.ref}
`
5. **VarDict**  
   - Ğ­Ñ„Ñ„ĞµĞºÑ‚Ğ¸Ğ²ĞµĞ½ Ğ´Ğ»Ñ Ğ³ĞµÑ‚ĞµÑ€Ğ¾Ğ³ĞµĞ½Ğ½Ñ‹Ñ… Ğ¾Ğ¿ÑƒÑ…Ğ¾Ğ»ĞµĞ¹.  
   - Ğ˜ÑĞ¿Ğ¾Ğ»ÑŒĞ·ÑƒĞµÑ‚ Random Forest.
 `#VarDict
 VarDict
	â€“dedup \
	-th 8 \
	-G {input.ref} \
	-f 0.02 \
	-N {wildcards.seq_name} \
	-r 3 \
	-b "{input.paired[0]}|{input.paired[1]}" \
	-c 1 -S 2 -E 3 -g 4 {input.target} | testsomatic.R | var2vcf_paired.pl -N "{wildcards.seq_name}|{wildcards.normal}" -f 0.02 > {output}` 
     
